--- origsrc/aften-0.0.8/CMakeLists.txt	2007-09-07 20:51:47.000000000 -0500
+++ src/aften-0.0.8/CMakeLists.txt	2009-11-11 01:25:49.186534800 -0600
@@ -136,6 +136,8 @@ IF(UNIX)
 
   IF(APPLE)
     ADD_DEFINE(SYS_DARWIN)
+  ELSEIF(CYGWIN)
+    ADD_DEFINE(SYS_CYGWIN)
   ELSE(APPLE)
     CHECK_FUNCTION_DEFINE("#include <sys/sysinfo.h>" "get_nprocs" "()" HAVE_GET_NPROCS)
 
--- origsrc/aften-0.0.8/libaften/threading.h	2007-07-08 00:11:52.000000000 -0500
+++ src/aften-0.0.8/libaften/threading.h	2009-11-11 01:27:23.496665000 -0600
@@ -92,6 +92,14 @@ get_ncpus()
 
     return sysctl(mib, 2, &numCPUs, &len, NULL, 0) ? 1 : numCPUs;
 }
+#elif defined(SYS_CYGWIN)
+#include <unistd.h>
+
+static inline int
+get_ncpus()
+{
+    return sysconf(_SC_NPROCESSORS_ONLN);
+}
 #else
 
 static inline int
